cookbooks/child-workflows/child-workflows-megafood.md

```diff
@@ -0,0 +1,321 @@
+
+### What are child workflows?
+
+Cadence's core abstraction is a fault-oblivious stateful **workflow**. A workflow may start other workflows and wait for them to complete, either synchronously or asynchronously. These are known as **child workflows**, and are a powerful tool which enable developers to coordinate large and complex tasks within the Cadence framework.
+
+Workflows evolve over time and child workflows may develop as a result of that evolutuion, such as when two existing workflow processes are joined, or designed from the start to take advantage the features available in Cadence.
+
+![Child workflow diagram](images/TBA.png)
+
+*Fig 1. Flow diagram for a child workflow process*
+
+### What are the advantages of child workflows?
+
+Child workflows have similar functionality to **activities**. An activity is a business-level function that implements application logic such as calling a service or transcoding a media file. 
+The parent workflow can monitor the child worflow and track its progress, and signals can be pass back and forth between the two with client SDK methods.
+
+Child workflows can be implemented by separate **worker processes**. A worker process is the process which is invoked to execute a particular workflow. 
+Separating out a set of activities into a child workflow allows you to ensure the code that gets executed the most often has the most resources by scaling those workers appropriately. 
+
+Another advantage of using child workflows is to work around the built in Cadence limits. Cadence limits the number of activities that a single workflow can execute. The limits were developed to ensure the most efficient operation of a Cadence cluster.
+For workflows running a large number of activities, we can use child workflows to distribute the work, thereby removing the limit and also gaining the other advantages mentioned.
+
+### When should I use a child workflow?
+
+In many ways, the answer to this question is similar to "when should I extract this code into a function?". If you find your workflows calling the same set of activities, in the same order, it is a prime candidate to be invoked via child workflow.
+
+If your workflow is approaching the limit of activity invocations, using child workflows is required to avoid running into the limits previously mentioned.
+
+Finally, child workflows are a great way to set boundaries between areas of functionality. Keeping all the code related to a specific functional area or responsibility makes them more reusable. We will see an example of this in the example below, where we use child workflows to encapsulate the workflows of different restaurants and delivery processes.
+
+### Invoking a child workflow
+
+The code to invoke a child workflow is almost identical to invoking a regular workflow:
+```java
+    // We first create a stub for the child workflow and include a set of options
+    SubWorkflow child = Workflow.newChildWorkflowStub(SubWorkflow.class, options);
```

Comment:
Is `SubWorkflow` a Cadence API thing, or the interface for the child workflow?  Maybe a batter name would be something that uses "child" rather than "sub" (since we're using the term child workflow everywhere else). e.g `MyCustomChildWorkflow`